src/test/java/seedu/address/logic/parser/IndexIdPairTest.java

```diff
@@ -0,0 +1,313 @@
+import seedu.address.model.person.Person;
+
+public class IndexIdPairTest {
+    private String personIdString = " idp/S123";
+    private String locationIdString = " idl/L123";
+    private String indexString = "1";
+    private String outOfBoundIndex = "2";
+    private String invalidIndex = "0";
+    private String invalidId = " idp/   ";
+    private String empty = "";
+    private String personIdNotExistString = " idp/not exist";
+    private String locationIdNotExistString = " idl/not exist";
+    private Id personId = new Id("S123");
+    private Id locationId = new Id("L123");
+    private Id idNotExist = new Id("not exist");
+
+    private ArgumentMultimap argMultimap;
+    private IndexIdPair indexIdPair;
+    private ModelStubForPair modelStub = new ModelStubForPair(DEFAULT_LOCATION, DEFAULT_PERSON);
+
+    @Test
+    public void checkIndexOrIdOnly_indexOrIdOnly_true() {
+        // person Id only
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertTrue(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+
+        // location Id only
+        argMultimap = ArgumentTokenizer.tokenize(locationIdString, PREFIX_LOCATION_ID);
+        assertTrue(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+    }
+
+    @Test
+    public void checkIndexOrIdOnly_indexAndIdOrNeither_false() {
+        // person Id and person Index
+        argMultimap = ArgumentTokenizer.tokenize(indexString + personIdString, PREFIX_PERSON_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+
+        // location Id and location Index
+        argMultimap = ArgumentTokenizer.tokenize(indexString + locationIdString, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+
+        // empty string
+        argMultimap = ArgumentTokenizer.tokenize(empty, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+    }
+
+    @Test
+    public void checkIndexOrIdOnly_wrongPrefix_false() {
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_LOCATION_ID));
+
+        argMultimap = ArgumentTokenizer.tokenize(locationIdString, PREFIX_LOCATION_ID);
+        assertFalse(IndexIdPair.checkIndexOrIdOnly(argMultimap, PREFIX_PERSON_ID));
+    }
+
+    @Test
+    public void constructor_nullValues_throwNullPointerException() {
+        argMultimap = ArgumentTokenizer.tokenize(personIdString, PREFIX_PERSON_ID);
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(null, null));
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(null, PREFIX_PERSON_ID));
+        assertThrows(NullPointerException.class, () -> new IndexIdPair(argMultimap, null));
+    }
+
+    @Test
+    public void constructor_validArgs_success() {
```

Comment:
Might need to add the corresponding tests for constructor related to the comment in the IndexIdPair class.

Reply:
Got it! Done!